### PR TITLE
Feature/configurable http redirect code in kubernetes

### PR DESCRIPTION
**Pull request by aryszka**
configurable status code for the Kubernetes https redirect. Addressing: https://github.com/zalando/skipper/issues/794

allows to:
- [x] change the default redirect code
- [x] enable the redirect only for specific ingresses, when it's globally disabled
- [x] disable the redirect only for specific ingresses, when it's globally enabled
- [x] override the redirect code for specific ingresses, when it's globally enabled and not disabled for the ingress, or it's globally disabled and enabled for the ingress
- [x] documentation

Ingresses currently cannot override the redirect behavior for the routes generated for the ingress default backends. The reason for this is that apparently there's a conflict with these routes. This conflict seems to be an already existing bug, and will be separately investigated. The redirect handling of the default backends needs to be revisited after that. (See https://github.com/zalando/skipper/issues/801)

### CODE REVIEW

File: dataclients/kubernetes/httpsredirect_test.go [hunk 1]
```diff
@@ -197,3 +206,468 @@ func TestNoHTTPSRedirect(t *testing.T) {
...
+	for _, i := range []int{3, 4, 5, 7} {
+		expect[i].Shunt = false
+	}
+
+	sort.Sort(sortRoutes(r))
+	sort.Sort(sortRoutes(expect))
+
+	diff := cmp.Diff(r, expect)
+	if diff != "" {
+		t.Error(diff)
```
Comment (szuecs) [inline dataclients/kubernetes/httpsredirect_test.go]: :+1:

File: dataclients/kubernetes/kube.go [hunk 2]
```diff
@@ -845,8 +857,21 @@ func (c *Client) ingressToRoutes(items []*ingressItem) ([]*eskip.Route, error) {
 			"ingress": fmt.Sprintf("%s/%s", i.Metadata.Namespace, i.Metadata.Name),
 		})
 
+		redirect.initCurrent(i.Metadata)
+
 		if r, ok, err := c.convertDefaultBackend(i); ok {
 			routes = append(routes, r...)
+
+			// we cannot control the redirect for the default backends, because
```
Comment (szuecs) [inline dataclients/kubernetes/kube.go]: Can you link the issue or comment in an issue you have for this commented code, too?

Comment (aryszka) [reply dataclients/kubernetes/kube.go]: https://github.com/zalando/skipper/issues/801

**Comment by szuecs** [comment]
:+1:

**Comment by aryszka** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Pull request merged**